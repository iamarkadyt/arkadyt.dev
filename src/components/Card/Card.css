.Card-container {
    width: 470px;
    display: grid;
    grid-template-rows: 1fr 1fr;
    box-shadow: 0 2px 20px 0 rgba(0,0,0,0.15);
}

.Card-container > a {
    display: block;
}

.Card-container > a > img {
    min-height: 250px;
    max-height: 250px;
    width: 100%;
    object-fit: cover;
}

.Card-content {
    padding: 45px;
}

.Card-content > h3 {
    font-family: 'Montserrat', sans-serif;
    margin: 0;
}

.Card-content > span {
    font-family: 'Open Sans', sans-serif;
    display: block;
    margin-top: 20px;
}

.Card-tags {
    font-family: 'Open Sans', sans-serif;
    margin-top: 20px;
}

.Card-tag {
    display: inline-block;
    background-color: #487eb0;
    color: white;
    padding: 2px 15px;
    margin-right: 10px;
}