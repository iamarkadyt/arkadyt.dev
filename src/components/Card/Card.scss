@import 'variables';

.Card-container {
    width: 47rem;
    display: grid;
    grid-template-rows: 1fr 1fr;
    box-shadow: 0 .2rem 2rem 0 rgba(0, 0, 0, 0.15);
    a {
        display: block;
        img {
            min-height: 25rem;
            max-height: 25rem;
            width: 100%;
            object-fit: cover;
        }
    }
    .description {
        padding: 4.5rem;
        h3 {
            font-family: 'Montserrat', sans-serif;
            font-size: 2rem;
            margin: 0;
        }
        >span {
            font-family: 'Open Sans', sans-serif;
            font-size: 1.6rem;
            display: block;
            margin-top: 2rem;
        }
        .tags {
            font-family: 'Open Sans', sans-serif;
            font-size: 1.6rem;
            margin-top: 2rem;
            span {
                display: inline-block;
                background-color: $color-accent-primary;
                color: $color-primary-white;
                padding: .2rem 1.5rem;
                margin-right: 1rem;
            }
        }
    }
}