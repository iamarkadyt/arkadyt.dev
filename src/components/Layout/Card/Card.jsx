import React from 'react';
import { Route, Switch, Redirect } from 'react-router-dom';
import DotNav from '../DotNav/DotNav';
import classes from './Card.module.css';

const card = props => {
    return <div className={classes.container}>
        <div className={classes.grid}>
            <Switch>
                {props.images.map((url, index) => {
                    return <Route key={index} path={`${props.match.url}/${index}`} render={() =>
                        <React.Fragment>
                            <img src={url} alt='' className={classes.background} />
                            <img src={url} alt='' className={classes.widget} />
                        </React.Fragment>
                    } />
                })}
                <Redirect to={`${props.match.url}/0`} />
            </Switch>
            <DotNav images={props.images} />
            <div className={classes.description}>
                <h1 className={classes.header}>{props.header}</h1><br />
                <p>{props.body}</p><br />
                {props.tags.map((item, index) => {
                    return <span key={index} className={classes.tag}>{item}</span>;
                })}<br />
                {'youtubeUrl' in props ? 
                    <a href={props.youtubeUrl} className={[classes.tag, classes.youtube].join(' ')}>
                        View on Youtube</a> : null}
                <a href={props.url} className={[classes.tag, classes.github].join(' ')}>
                    View on GitHub</a>
            </div>
        </div>
    </div>;
};

export default card;