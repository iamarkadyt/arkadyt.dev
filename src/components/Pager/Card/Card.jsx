import React from 'react';
import classes from './Card.module.css';
import image from './RD3CTFQ3_11.jpg';

const card = props => {
    return <div className={classes.container}>
        <img src={image} alt='' className={classes.widget} />
        <div className={classes.description}>
            <h1 className={classes.header}>{props.header}</h1><br />
            <p>{props.body}</p><br />
            <a href={props.url}>See project</a>
        </div>
    </div>;
};

export default card;