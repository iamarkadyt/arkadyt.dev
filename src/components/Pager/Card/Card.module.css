.container {
    height: 100%;
    width: 100%;
}

.container img.background {
    position: fixed;
    z-index: -50;
    top: -3%;
    left: -3%;
    width: 106%;
    height: 106%;
    object-fit: cover;
    filter: blur(5px) brightness(50%);
    backface-visibility: hidden;
}

.container .grid {
    width: 100%;
    height: 100%;
    display: grid;
    grid-template-columns: 60% 40%;
    grid-template-rows: 80% 20%;
}

.container img.widget {
    grid-column: 1/2;
    grid-row: 1/2;
    width: 95%;
    height: 95%;
    margin: 5% 0 0 5%;
    object-fit: cover;
    box-shadow: 0 0 25px black;
}

.description {
    grid-column: 2/3;
    grid-row: 1/3;
    align-self: end;
    text-align: right;
    color: #eee;
    margin: 25px;
}

.description p {
    font-family: 'Paragraph', sans-serif;
    font-size: 1.1em;
    pointer-events: none;
}

.description .header {
    font-family: 'Header', sans-serif;
    font-size: 3em;
    text-transform: uppercase;
    margin: 0;
    pointer-events: none;
}

.description a {
    font-family: 'Paragraph', sans-serif;
    color: #eee;
    text-decoration: none;
    background-color: green;
    padding: 4px 32px;
    border-radius: 4px;
}

.description .tag {
    display: inline-block;
    font-family: 'Paragraph', sans-serif;
    color: #eee;
    background-color: darkcyan;
    padding: 4px 32px;
    margin: 0 0 15px 5px;
    border-radius: 4px;
    pointer-events: none;
}

@font-face {
    font-family: 'Header';
    src: url(../../../fonts/steelfishrg.ttf) format('truetype');
}

@font-face {
    font-family: 'Paragraph';
    src: url(../../../fonts/dustismo.ttf) format('truetype');
}