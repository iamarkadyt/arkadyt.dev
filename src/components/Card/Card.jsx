import React from 'react';
import './Card.css';
import IconHref from 'react-icons/lib/fa/external-link';

const card = props => {
    return <div className="Card-container">
        <div className="frontface">
            <img src={props.image} alt='' />
            <div className="description">
                <h3>{props.title}</h3>
                <span>{props.description}</span>
                <div className="tags">
                    {props.tags.map((item, index) => (
                        <span key={index}>{item}</span>
                    ))}
                </div>
            </div>
            <div className="hint"><IconHref /></div>
        </div>
        <div className="backface">
            {props.links.map((item, index) => {
                return <a key={index} href={item.href} target="_blank">
                    {item.title}&nbsp;&nbsp;<IconHref />
                </a>
            })}
        </div>
    </div>;
};

export default card;