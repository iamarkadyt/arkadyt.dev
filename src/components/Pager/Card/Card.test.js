import React from 'react';
import Enzyme, { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Card from './Card';

configure({ adapter: new Adapter() });

it('matches snapshot.', () => {
    const tree = renderer.create(
        <Card url='/url' header='Header' img='/img-link'>
            <div className="sampleDiv"></div>
        </Card>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});