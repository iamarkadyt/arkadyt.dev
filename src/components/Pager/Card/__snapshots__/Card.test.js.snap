// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot. 1`] = `
<div
  className="container"
>
  <img
    alt=""
    src="RD3CTFQ3_11.jpg"
  />
  <div
    className="description"
  >
    <h1
      className="header"
    >
      Header
    </h1>
    <p />
    <a>
      /url
    </a>
  </div>
</div>
`;
